src/Foundatio.Extensions.Hosting/Jobs/JobHostExtensions.cs

```diff
@@ -55,18 +59,19 @@ public static IServiceCollection AddJob(this IServiceCollection services, Func<I
             return services.AddTransient<IHostedService>(s =>
             {
                 if (jobOptions.JobFactory == null)
-                    jobOptions.JobFactory = s.GetRequiredService<T>;
+                    jobOptions.JobFactory = _ => s.GetRequiredService<T>();
 
                 return new HostedJobService(s, jobOptions, s.GetService<ILoggerFactory>());
             });
         }
-        else
-        {
-            if (!services.Any(s => s.ServiceType == typeof(IHostedService) && s.ImplementationType == typeof(ScheduledJobService)))
-                services.AddTransient<IHostedService, ScheduledJobService>();
 
-            return services.AddTransient(s => new ScheduledJobRegistration(jobOptions.JobFactory ?? (s.GetRequiredService<T>), jobOptions.CronSchedule));
-        }
+        if (!services.Any(s => s.ServiceType == typeof(IHostedService) && s.ImplementationType == typeof(ScheduledJobService)))
+            services.AddTransient<IHostedService, ScheduledJobService>();
+
+        if (!services.Any(s => s.ServiceType == typeof(IScheduledJobManager) && s.ImplementationType == typeof(ScheduledJobManager)))
+            services.AddSingleton<ScheduledJobManager>();
```

Comment:
Can we share any of this logic or move it to a private implementation / extension?

Reply:
It's like 4 lines of repeated code. Not worried about it.